packages/hyperion-flowlet/src/Flowlet.ts

```diff
@@ -2,14 +2,23 @@
  * Copyright (c) Meta Platforms, Inc. and affiliates. All Rights Reserved.
  */
 
+import { Hook } from "@hyperion/hook";
+
+export const onFlowletInit = new Hook<(flowlet: Flowlet) => void>();
+
+let flowletID: number = 0;
+
 export class Flowlet<T extends {} = {}> {
   readonly data: T;
+  private _id: number = flowletID++;
```

Comment:
instead you could just say
```
readonly id: number = flowletID++;
```

That way, the value stays fixed and you don't need a getter to protect the value.